Sources/MapboxSearch/PublicAPI/Engine/CategorySearchEngine.swift

```diff
@@ -24,7 +24,12 @@ public class CategorySearchEngine: AbstractSearchEngine {
             categories: [categoryName],
             options: coreOptions
         ) { [weak self, weak eventsManager] coreResponse in
```

Comment:
Our alternative seems to be to capture `self` strongly, creating a reference cycle for the time of the response. Have you considered this approach?